packages/replay/src/integration.ts

```diff
@@ -38,27 +39,57 @@ export class Replay implements Integration {
     flushMaxDelay = DEFAULT_FLUSH_MAX_DELAY,
     stickySession = true,
     useCompression = true,
+    _experiments = {},
     sessionSampleRate,
     errorSampleRate,
     maskAllText,
-    maskTextSelector,
     maskAllInputs = true,
     blockAllMedia = true,
-    _experiments = {},
-    blockClass = 'sentry-block',
-    ignoreClass = 'sentry-ignore',
-    maskTextClass = 'sentry-mask',
-    blockSelector = '[data-sentry-block]',
-    ..._recordingOptions
+
+    mask = [],
+    unmask = [],
+    block = [],
+    unblock = [],
+    ignore = [],
+    maskFn,
+
+    // eslint-disable-next-line deprecation/deprecation
+    blockClass,
+    // eslint-disable-next-line deprecation/deprecation
+    blockSelector,
+    // eslint-disable-next-line deprecation/deprecation
+    maskTextClass,
+    // eslint-disable-next-line deprecation/deprecation
+    maskTextSelector,
+    // eslint-disable-next-line deprecation/deprecation
+    ignoreClass,
   }: ReplayConfiguration = {}) {
     this._recordingOptions = {
       maskAllInputs,
-      blockClass,
-      ignoreClass,
-      maskTextClass,
-      maskTextSelector,
-      blockSelector,
-      ..._recordingOptions,
+      maskTextFn: maskFn,
+      maskInputFn: maskFn,
+
+      ...getPrivacyOptions({
+        mask,
+        unmask,
+        block,
+        unblock,
+        ignore,
+        blockClass,
+        blockSelector,
+        maskTextClass,
+        maskTextSelector,
+        ignoreClass,
+      }),
+
+      // Our defaults
+      slimDOMOptions: 'all',
+      inlineStylesheet: true,
+      // Disable inline images as it will increase segment/replay size
+      inlineImages: false,
+      // collect fonts, but be aware that `sentry.io` needs to be an allowed
+      // origin for playback
+      collectFonts: true,
```

Comment:
We may get some CORS errors on playback, but it should be fine as it will fallback to system fonts.